src/parser.cc

```diff
@@ -1942,27 +1942,17 @@ void copyChildren(ElementPtr _sdf,
       ElementPtr element(new Element);
       element->SetParent(_sdf);
       element->SetName(elemName);
-      std::optional<std::string> typeName = std::nullopt;
-      for (const tinyxml2::XMLAttribute *attribute = elemXml->FirstAttribute();
-           attribute; attribute = attribute->Next())
+      if (elemXml->GetText() != nullptr)
       {
-        const std::string attributeName(attribute->Name());
-        // TODO(chapulina) This isn't a good place to parse pose-specific
-        // attributes
-        if (elemName == "pose" && attributeName == "type")
-          typeName = attribute->Value();
-
-        element->AddAttribute(attributeName, "string", "", 1, "");
-        element->GetAttribute(attributeName)->SetFromString(
-          attribute->Value());
+        element->AddValue("string", elemXml->GetText(), true);
       }
 
-      if (elemXml->GetText() != nullptr)
+      for (const tinyxml2::XMLAttribute *attribute = elemXml->FirstAttribute();
+           attribute; attribute = attribute->Next())
       {
-        if (typeName.has_value())
-          element->AddValue(typeName.value(), elemXml->GetText(), true);
-        else
-          element->AddValue("string", elemXml->GetText(), true);
+        element->AddAttribute(attribute->Name(), "string", "", 1, "");
+        element->GetAttribute(attribute->Name())->SetFromString(
+            attribute->Value());
```

Comment:
it looks like you've reversed the order of some logic in parser.cc, calling `element->AddValue` before / after setting the attribute values. Does this order matter? If not, can you reverse it so that the diff will be easier to read?

Reply:
This order does not matter as in this case the value is always added as type `string`.

Only with specific types like `pose`, where the attributes like `degrees` or `rotation_format` are read, will it matter. But it still makes sense to have attributes handled before the value, regardless.

Reversed in https://github.com/ignitionrobotics/sdformat/pull/811/commits/c630d3093775b04418eae131e2fbb9230a14dbd7.